### PR TITLE
[sonic-yang] Backlinks are none when there is 0 backlinks

### PR DESCRIPTION
**Pull request by ghooo**
#### Why I did it
Backlinks are none when there is 0 backlinks. without this change `backlinks.number()` throws an exception when backlinks==None.

#### How I did it
Allow backlinks to be None, so we get an empty list as a return from find_data_dependencies

#### How to verify it
```python
    def test_find_leaf_dependencies_no_schema_dependencies(self):
        sy = sonic_yang.SonicYang(YANG_DIR)
        sy.loadYangModel()
        sy.loadData(json.loads("""
{
    "PORT": {
        "Ethernet0": {
            "alias":"eth0",
            "lanes": "65",
            "speed": "100000"
        }
    }
}
            """))

        # the PORT alias has no leafrefs pointing to it
        xpath = "/sonic-port:sonic-port/PORT/PORT_LIST[name='Ethernet0']/alias"
        deps = sy.find_data_dependencies(xpath)

        self.assertEqual(0, len(deps))
```
(added UT for the same idea)

### CODE REVIEW

**Comment by qiluo-msft** [comment]
@pchaudhary Could you help review

**Comment by qiluo-msft** [comment]
@praveen-li Could you help review

**Review by qiluo-msft — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Review by praveen-li — APPROVED** [review]
LGTM, fix the alignment if wrong, before merge. Thx.

**Pull request merged**
